pulpcore/pulpcore/app/serializers/repository.py

```diff
@@ -190,6 +190,28 @@ class Meta:
         ]
 
 
+class HyperlinkedRelatedPublicationField(serializers.HyperlinkedRelatedField):
+    """
+    Avoids the API consumer passing an incomplete Publication to a Serializer.
+    """
+    queryset = models.Publication.objects.all()
```

Comment:
I think a similar effect can be reached by `queryset = models.Publication.objects.exclude(complete=False)` without overriding `get_object`

Reply:
ack; I'm all in for reducing the commit size :D